server/src/services/analyticsService.ts

```diff
@@ -0,0 +1,55 @@
+import { CheckInModel } from "../models/checkInModel";
+import { User } from "../models/userModel";
+
+export class AnalyticsService {
+  static async getAllCheckInUsers() {
+    try {
+      const allUsers = await CheckInModel.find();
```

Comment:
you can just pass in params to return all users which checked is false instead of the for each loop